packages/tooltip/src/tooltipUtils.tsx

```diff
@@ -0,0 +1,106 @@
+import { css } from '@leafygreen-ui/emotion';
+
+export function trianglePosition(
+  alignment: string,
+  justification: string,
+  triggerRect: DOMRect | ClientRect | null,
+) {
+  if (!alignment || !justification || !triggerRect) {
+    return '';
+  }
+
+  const borderOffset = 'calc(100% - 1px)';
+
+  const containerSize = 15;
+  const notchSize = 8;
+  const containerOffsetX = (triggerRect.width - containerSize) / 2;
+  const containerOffsetY = (triggerRect.height - containerSize) / 2;
+  const notchOverlap = -notchSize / 2;
+
+  type Styles = 'left' | 'right' | 'top' | 'bottom' | 'margin';
+  const notchStyleObj: { [K in Styles]?: string } = {};
+  const containerStyleObj: { [K in Styles]?: string } = {};
+
+  switch (alignment) {
```

Comment:
Might just be a preference of mine, but rather than nesting switch statements, I'd prefer we use a good ol' conditional for the outer switch here.

Reply:
Talked offline, leaving as-is.